### PR TITLE
Explicitly install transitive dependencies

### PR DESCRIPTION
**Pull request by joecowton1**
## What does this change?

Explicitly installs each of the transitive dependencies we use in DCR.

## Why?

As part of the work we've been doing to unpick the dependency tangle that existed across this repo, we've repeatedly come across issues where transitive dependencies (those installed by other packages) are masking issues: for example, Storybook installs a number of packages that are required to run DCR in production; if we remove Storybook then the project won't build.

By explicitly installing each of the dependencies we move away from this situation somewhat. This also takes us closer to a place where we could switch package manager to pnpm and move DCR into the csnx monorepo. While both of these are still some way off, the closer DCR is to a state in which we _can_ do these things, the less blockers we'll encounter if and when the decision is made to move it.

### CODE REVIEW

File: dotcom-rendering/package.json [hunk 1]
```diff
@@ -175,47 +184,59 @@
...
 		"inquirer": "^8.2.4",
 		"jest": "^24.9.0",
 		"jest-environment-jsdom-sixteen": "^1.0.3",
 		"jest-teamcity-reporter": "^0.9.0",
 		"jsdom": "^20.0.0",
 		"lint-staged": "^12.3.4",
 		"load-json-file": "^6.2.0",
+		"lodash": "^4.17.21",
 		"lodash.debounce": "^4.0.8",
 		"lodash.get": "^4.4.2",
```
Comment (mxdvl) [inline dotcom-rendering/package.json:204]: Gotta love the fact that we now have two flavours of `lodash`!

**Review by mxdvl — APPROVED** [review]
This makes sense, and makes explicit what has been implicitly happening for a while.

---

I know @jamesgorrie was worried about the ever increasing size of our node_modules, and we did have an instance last year where we filled our production instances’ disks so much that they could no longer start, but I don’t think there’s any meaningful distinction we could make between DEV and PROD dependencies.

File: dotcom-rendering/package.json (hunk 1 above)
Comment (joecowton1) [reply dotcom-rendering/package.json:204]: Three even 🤯

Comment (mxdvl) [reply dotcom-rendering/package.json:204]: Well the `lodash.*` are tree-shaken ESM versions of the main one… so in my mind they are the same “flavour”.

Comment (mxdvl) [reply dotcom-rendering/package.json:204]: This actually caused an issue, which is fixed by #8761

**Pull request merged**